specifyweb/frontend/js_src/lib/components/Router/RouterUtils.tsx

```diff
@@ -126,18 +126,20 @@ export const toReactRoutes = (
  * when any component is being loaded.
  */
 export function Async({
-  Element,
   title,
+  ...rest
 }: {
-  readonly Element: React.FunctionComponent;
-  readonly title: LocalizedString | undefined;
-}): JSX.Element {
+  readonly title?: LocalizedString | undefined;
+} & (
+  | { readonly Element: React.FunctionComponent }
+  | { readonly Node: React.ReactNode }
```

Comment:
why not just `Element: React.FunctionComponent | React.ReactNode`?

and then in the code do `typeof Element === 'function'` to differentiate

Reply:
Hmm both of them are of type objects. I experimented with putting in above function definition
```
  console.log(
    Object.keys(rest)[0],
    typeof ('Element' in rest ? rest.Element : rest.Node)
  );
```
and got 'object' for both of them (Element, and Node). I suspect React handles conversion for us when actually calling it from lazy.